### DISCUSSION TITLE
Flux in two dimensions using sympy.vector.vector_integrate

### DISCUSSION THREAD

**Discussion by Morten10** [sympy/sympy#24124] [Q&A]
edit:
I've asked the same question at stackoverflow and apparently SymPy doesn't consider the normal vector to the circle. 
[stack](https://stackoverflow.com/questions/73997770/flux-in-two-dimensions-using-pyhton-sympy-vector-vector-integrate?noredirect=1#comment130656701_73997770)
how can it be fixed?

I am having trouble calculating the following example
https://www.khanacademy.org/math/multivariable-calculus/integrating-multivariable-functions/line-integrals-in-vector-fields-articles/a/flux-in-two-dimensions
where F is 
![image](https://user-images.githubusercontent.com/115358235/194711492-d7940252-df9b-40c9-af82-3e61d5a1fc8f.png)
and r(t) is
![image](https://user-images.githubusercontent.com/115358235/194711516-bdb02bcf-4a66-4910-95db-68918877d9e0.png)

```
from sympy import sin, cos, exp, pi, symbols
from sympy.vector import CoordSys3D, ParametricRegion, ImplicitRegion, vector_integrate
from sympy.abc import r, x, y, z, theta, phi,a
C = CoordSys3D('C')
F=C.x**2*C.i+C.y*C.j
display('F=',F)
semicircle = ParametricRegion((3*cos(theta), 3*sin(theta)), (theta, 0, 2*pi))
flux = vector_integrate(F, semicircle)
display('flux=',flux)
```

**Comment by oscarbenjamin** [comment]
GitHub issues are not for getting help with using SymPy. Please use the mailing list for that.

**Comment by Davide-sd** [comment]
@oscarbenjamin I believe this is a proper issue.

[The documentation](https://docs.sympy.org/latest/modules/vector/vector_integration.html) states that it is possible to compute the flux with `vector_integrate`. However, the example shown by @Morten10 produces the wrong result.

I have no idea how many "flux definitions" are possible, but looking at the source code it is evident that whatever is computed by SymPy is different from the definition provided in KhanAcademy. This is what the articles compute:

![Screenshot 2022-10-09 at 10-19-01 Flux in two dimensions (article) Khan Academy](https://user-images.githubusercontent.com/9921777/194745887-2bb5a22e-53ca-4062-8df5-57d77ea9795a.png)

![Screenshot 2022-10-09 at 10-18-26 Flux in two dimensions (article) Khan Academy](https://user-images.githubusercontent.com/9921777/194745903-9883d5f5-6f6a-42f8-962d-cea5da851dd3.png)

Note the dot product between the vector field and the normal vector to the parametric region.

Instead, [this is what SymPy computes](https://github.com/sympy/sympy/blob/master/sympy/vector/integrals.py#L77): normal vector to the parametric region never gets computed.

So, what is SymPy computing? And is the documentation correct in stating that it is able to compute the flux?

**Reply by oscarbenjamin** [reply]
The flux example in the docs uses a 2D surface in 3D space whereas the OP example asks to integrate over a 1D curve in 2D space.
